Core/Logging.swift

```diff
@@ -18,62 +18,48 @@
 //
 
 import Foundation
-import os.log
 import BrowserServicesKit
 import Common
 
 public extension OSLog {
-    
-    static var generalLog: OSLog {
-        Logging.generalLoggingEnabled ? Logging.generalLog : .disabled
-    }
-    
-    static var contentBlockingLog: OSLog {
-        Logging.contentBlockingLoggingEnabled ? Logging.contentBlockingLog : .disabled
-    }
 
-    static var adAttributionLog: OSLog {
-        Logging.adAttributionLoggingEnabled ? Logging.adAttributionLog : .disabled
-    }
-    
-    static var lifecycleLog: OSLog {
-        Logging.lifecycleLoggingEnabled ? Logging.lifecycleLog : .disabled
-    }
-    
-    static var autoconsentLog: OSLog {
-        Logging.autoconsentLoggingEnabled ? Logging.autoconsentLog : .disabled
+    enum AppCategories: String, CaseIterable {
+        case generalLog = "DDG General"
+        case contentBlockingLog = "DDG Content Blocking"
+        case adAttributionLog = "DDG AdAttribution"
+        case lifecycleLog = "DDG Lifecycle"
+        case autoconsentLog = "DDG Autoconsent"
+        case configurationLog = "DDG Configuration"
     }
 
-    static var configurationLog: OSLog {
-        Logging.configurationLoggingEnabled ? Logging.configurationLog : .disabled
-    }
+    @OSLogWrapper(.generalLog) static var generalLog
+    @OSLogWrapper(.contentBlockingLog) static var contentBlockingLog
+    @OSLogWrapper(.adAttributionLog) static var adAttributionLog
+    @OSLogWrapper(.lifecycleLog) static var lifecycleLog
+    @OSLogWrapper(.autoconsentLog) static var autoconsentLog
+    @OSLogWrapper(.configurationLog) static var configurationLog
+
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    // To activate Logging Categories add categories here:
+    static var enabledCategories: Set<AppCategories> = [
+        .generalLog,
+        .contentBlockingLog,
+        .adAttributionLog,
+        .lifecycleLog,
+        .configurationLog
+    ]
 
 }
 
-struct Logging {
+public extension OSLog.OSLogWrapper {
 
-    fileprivate static let generalLoggingEnabled = true
```

Comment:
@tomasstrba, these logs were originally enabled in the main branch

Reply:
yeah, makes sense! Thanks! :)